### DISCUSSION TITLE
Shopify Shop with 200 products (2000 - 3000 variants) too much?

### DISCUSSION THREAD

**Discussion by fnebenfuehr** [gatsbyjs/gatsby#30840] [Help]
Hi there!

We are using gatsby and shopify-source-plugin to build an e-commerce shop. The shop has approximately 200 products and 2000 - 3000 variants. We launched the shop last month on gatsby cloud using gatsbyjs 2.0. Since we had issues with cloud builds on the gatsby cloud platform we are now migrating to gatsbyjs 3.0. Unfortunately updating gatsbyjs made our builds crash (memory exceeding limit). We tried several other platforms like Netlify or Vercel without success. Upgrading the builds plan to the performance plan fixes the issue temporarily. So I am wondering if we are doing something horribly wrong or if we choose the wrong framework/platform for our purpose? Is the shop too large for a static site generator? Even if the builds are now running through on the performance plan. Paying 85$ a month for just building is not in any relation to using a shopify theme. Unfortunately, we can't sell the (great) developer experience to the client. Certainly not if they are waiting several minutes for changes to reflect and pay a lot of money. I would be very happy on any thoughts from the community as well as the gatsby team?

Thanks,
Florian

**Comment by reco** [comment]
Good question. Would be amazing to get some guidance from the team!

+1

**Comment by KyleAMathews** [comment]
You might want to check out the coming new version of the shopify plugin as it leverages the shopify image cdn to avoid downloading and processing images https://www.gatsbyconf.com/event/announcing-gatsby-shopify-source-plugin-v2-release-candidate/
